### PR TITLE
Enable PHP 7.2-8.1 compat

### LINKED ISSUE #73: Future of this project
What is the future of this project? In my opinion, it is feature complete and only needs to have the latest versions of PHP (7.3, 7.4 & 8.0 as of this writing) tested via travis or Github Actions to ensure it continues to work with the currently supported version of PHP. Will anyone be updating the CI tests for this project?

### PR DESCRIPTION
**Pull request by koriym**
Closes #73, #76

I picked #72 PR for 4.x. (originally targeted to 2.x branch)  The rest of the PRs are left as they are, since I don't think I can merge them right away.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,12 @@
     "suggest": {
         "ext-openssl": "OpenSSL generates the best secure CSRF tokens.",
         "ext-mcrypt": "Mcrypt generates the next best secure CSRF tokens.",
-        "ircmaxell/random-lib": "A Library For Generating Secure Random Numbers"
+        "ircmaxell/random-lib": "A Library For Generating Secure Random Numbers",
+        "paragonie/random_compat": "PHP 5.x polyfill for random_bytes() and random_int() from PHP 7"
+    },
```
Comment (harikt) [inline composer.json:32]: I think we can remove random_compact

File: composer.json [hunk 2]
```diff
@@ -28,7 +28,12 @@
...
+    "require-dev": {
+        "aura/di": "~2.0",
```
Comment (harikt) [inline composer.json]: Do we support di config in 4.x releases.

**Review by harikt — COMMENTED** [review]
I think we can remove aura di config test. The version itself has changed.

File: composer.json (hunk 2 above)
Comment (koriym) [reply composer.json]: Removed.

File: composer.json (hunk 1 above)
Comment (koriym) [reply composer.json:32]: Removed.

File: src/Randval.php [hunk 3]
```diff
@@ -59,7 +59,7 @@ public function generate()
         }
 
         if ($this->phpfunc->extension_loaded('mcrypt')) {
-            return $this->phpfunc->mcrypt_create_iv($bytes, MCRYPT_DEV_URANDOM);
+            return $this->phpfunc->random_bytes($bytes);
         }
```
Comment (harikt) [inline src/Randval.php:63]: I think we can get rid of this class

**Pull request merged**